selinon/dispatcher.py

```diff
@@ -225,3 +232,25 @@ def run(self, flow_name, node_args=None, parent=None, retried_count=None, retry=
             # Always an empty array.
             'active_nodes': state_dict.get('active_nodes', [])
         }
+
+    def _check_hanged_task(self, flow_info):
+        """A custom function to remove tasks which are rotating in dispatcher for more than given time.
+       :param flow_info: information about the current flow
+       """
+        node_args = flow_info['node_args']
+        if node_args is not None and 'flow_start_time' in node_args and node_args['flow_start_time'] is not None:
```

Comment:
`node_args.flow_start_time` doesn't look like a meta data managed by selinon. Which component sets that? Can you add more detailed description in the PR?

Reply:
@arajkumar it is added in last line of this function line number 256.
If a message is coming for the first time in dispatcher it is checking if key 'flow_start_time' is present then use it if not then consider current time as start time for flow and put it.
It can also be set from worker side when we initial a flow in Init Analysis Flow, but the messages which were stuck in queue were never coming back to worker side and they were picked and pushed back by this dispatcher class, that is why I had to add flow_start_time key from this class.

Reply:
by the way, hanged is an incorrect word. it should be "hung"

Reply:
Its changed now.